### ISSUE TITLE
`d3-brush::empty()` returns `true` incorrectly for one-dimensional brushes

### ISSUE THREAD

**Issue by ericsoco** [d3/d3#2928]
For one-dimensional brushes, [this logic](https://github.com/d3/d3-brush/blob/master/src/brush.js#L113-L116) seems incorrect:

```
function empty(extent) {
  return extent[0][0] === extent[1][0]
      || extent[0][1] === extent[1][1];
}
```

When passing a single pair of values to `move()` for a one-dimensional brush [as described in the docs](https://github.com/d3/d3-brush#brush_move), `__brush` ends up as `null` because of the `empty()` call [here](https://github.com/d3/d3-brush/blob/master/src/brush.js#L220).

Instead, `empty` should probably be

```
function empty(extent) {
  return extent[0][0] === extent[1][0]
      && extent[0][1] === extent[1][1];
}
```

for one-dimensional brushes.

**Comment by mbostock** [comment]
Are you basing this on a read of the code, or do you have a test case to assert that there’s a bug here? Because that *state*.selection is always two-dimensional, even for a one-dimensional brush; it is passed through *dim*.input, which substitutes the corresponding values from the brush’s extent for the other dimension (see [X](https://github.com/d3/d3-brush/blob/master/src/brush.js#L15) and [Y](https://github.com/d3/d3-brush/blob/master/src/brush.js#L22)).

**Comment by mbostock** [comment]
Also, I tested [this example](http://bl.ocks.org/mbostock/raw/6232537/) pasting in the following code to the console and it worked fine…

```js
d3.select(".brush").call(brush.move, [100, 200]);
```

**Comment by ericsoco** [comment]
Found this stepping through the code. I was passing a single pair of values into `move()`, and seeing `selection` set to `[0, val0], [0, val1]` for a `brushY()`. I believe the initial `extent()` is something like `[[0, 40], [0, 256]]`. Maybe this will work out if I let d3 calculate the default `extent()` instead (per our other thread)? Unfortunately can't check until tomorrow or later...

**Comment by ericsoco** [comment]
Re: the example, I was seeing a problem after calling `move()` when trying to get `brushSelection()` -- it was returning `null`. The `move()` call executed with no problems.

**Comment by mbostock** [comment]
Again in that example, this worked fine:

```js
d3.select(".brush").call(brush.move, [100, 200]);
d3.brushSelection(d3.select(".brush").node()); // [200, 300]
```

(Although in practice I rarely see the need for d3.brushSelection since the selection is available as d3.event.selection.)

**Comment by mbostock** [comment]
At any rate, to file a bug, please include some code I can run (either a unit test as a PR or a Gist on bl.ocks.org) that reliably demonstrates the issue. Thank you!

**Comment by ericsoco** [comment]
Ugh, I figured it out -- was passing `[[x0, x1], [y0, y1]]` to `extent()` instead of `[[x0, y0], [x1, y1]]`. Upper-left corner is `(0,0)`, so I was accidentally setting the width to 0, which results in a selection that is always empty no matter where it's `move()`d to. Sorry for the noise.

As for `brushSelection()`, I'm using it because I've decoupled my filtering code from my brush event handler, since multiple UI elements play into the filtering. So, `d3.event` doesn't necessarily point at the brush event, could be a different, more recent event. JIC you are interested in how the people are using the thing.

Thanks Mike!
